arch/arc/core/mpu/arc_mpu_v3_internal.h

```diff
@@ -656,7 +656,7 @@ int arc_core_mpu_get_max_domain_partition_regions(void)
 int arc_core_mpu_buffer_validate(void *addr, size_t size, int write)
 {
 	int r_index;
-
+	int key = irq_lock();
```

Comment:
You probably want arch_irq_lock() here if you're just checking local CPU state, or a spinlock if the state is visible from other CPUs.  Remember that in SMP, irq_lock() is a somewhat heavyweight **global** lock that will contend with any irq_lock() user anywhere in the system.

Reply:
thanks your comment, it make sense, i will modify it